corehq/apps/consumer_user/tests/test_views.py

```diff
@@ -0,0 +1,333 @@
+
+    def test_login_post(self):
+        email = 'log@log.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'patient_login_view-current_step': 'auth',
+        }
+        response = self.client.post(self.login_url, post_data)
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, self.homepage_url)
+
+    def test_login_accept_invitation(self):
+        email = 'log1@log1.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        invitation = ConsumerUserInvitation.objects.create(case_id='6', domain='1', invited_by='I',
+                                                           invited_on='2021-02-09 17:17:32.229524+00',
+                                                           email=email)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'patient_login_view-current_step': 'auth',
+        }
+        response = self.client.post(login_accept_url(invitation.id), post_data)
+        self.assertEqual(response.status_code, 302)
+        invitation.refresh_from_db()
+        self.assertTrue(invitation.accepted)
+        self.assertRedirects(response, self.homepage_url)
+
+
+class SignalTestCase(TestCase):
+
+    def setUp(self):
+        self.domain = 'consumer-invitation-test'
+
+    @override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
+    def test_method_send_email(self):
+        invitation_count = ConsumerUserInvitation.objects.count()
+        with patch('corehq.apps.hqwebapp.tasks.send_html_email_async.delay') as async_task, create_case(
+            self.domain,
+            'commcare-caseuserinvitation',
+            owner_id='comm_care',
+            update={'email': 'testing@testing.in'}
+        ) as case:
+            customer_invitation = ConsumerUserInvitation.objects.get(case_id=case.case_id, domain=case.domain)
+            self.assertEqual(customer_invitation.email, case.get_case_property('email'))
+            self.assertEqual(ConsumerUserInvitation.objects.count(), invitation_count + 1)
+            self.assertEqual(async_task.call_count, 1)
+            update_case(self.domain, case.case_id,
+                        case_properties={'contact_phone_number': '12345'})
+            self.assertEqual(ConsumerUserInvitation.objects.count(), invitation_count + 1)
+            self.assertEqual(async_task.call_count, 1)
+            update_case(self.domain, case.case_id,
+                        case_properties={'email': 'email@changed.in'})
+            self.assertEqual(ConsumerUserInvitation.objects.count(), invitation_count + 2)
```

Comment:
It would be useful to add a comment to each block here to explain what is going on. eg:
```
# Updating the case again with a changed email address creates a new invitation
```

Reply:
Added comments to that test